src/models/inbound/dfspLinking.model.ts

```diff
@@ -157,21 +196,95 @@ export class DFSPLinkingModel
 
     } catch (error) {
+      const mojaloopError = reformatError(error, this.logger)
       this.logger.push({ error }).error('requestIsValid -> consentRequestValidatedAndStored')
-      // TODO: PUT /consentRequest/{ID}/error if DFSP is unable to store consent Request or
-      //       send OTP
+
+      await this.thirdpartyRequests.putConsentRequestsError(
+        consentRequestsPostRequest.consentRequestId,
+        mojaloopError as unknown as fspiopAPI.Schemas.ErrorInformationObject,
+        toParticipantId
+      )
       // throw error to stop state machine
       throw error
     }
   }
 
+  async onSendLinkingChannelResponse (): Promise<void> {
+    const { consentRequestsPostRequest, consentRequestsIDPutRequest, toParticipantId} = this.data
+
+    const waitOnAuthTokenFromPISPResponseChannel = DFSPLinkingModel.notificationChannel(
+      DFSPLinkingPhase.waitOnAuthTokenFromPISPResponse,
+      consentRequestsPostRequest.consentRequestId
+    )
+    // now we send back a PUT /consentRequests/{ID}response to elicit
+    // a PATCH /consentRequests/{ID} response containing an authToken
+    const waitOnAuthToken = deferredJob(this.pubSub, waitOnAuthTokenFromPISPResponseChannel)
+      .init(async (channel) => {
+        const res = await this.thirdpartyRequests.putConsentRequests(
+          consentRequestsPostRequest.consentRequestId,
+          consentRequestsIDPutRequest,
+          toParticipantId
+        )
+
+        this.logger.push({ res, channel })
+          .log('ThirdpartyRequests.putConsentRequests call sent to auth service, listening on response')
+      })
+      .job(async (message: Message): Promise<void> => {
+        try {
+          type PutResponse =
+            tpAPI.Schemas.ConsentRequestsIDPatchRequest
+          type PutResponseOrError = PutResponse & fspiopAPI.Schemas.ErrorInformationObject
+          const putResponse = message as unknown as PutResponseOrError
+
+          if (putResponse.errorInformation) {
+            // if the PISP sends back any error, both machines will now
+            // need to be in an errored state
+            // store the error so we can transition to an errored state
+            this.data.errorInformation = putResponse.errorInformation as unknown as fspiopAPI.Schemas.ErrorInformation
+
+          } else {
+            this.data.consentRequestsIDPatchResponse = { ...message as unknown as PutResponse }
+          }
+        } catch (error) {
+          this.logger.push(error).error('ThirdpartyRequests.putConsentRequests request error')
+          return Promise.reject(error)
+        }
+      })
+      // since the PATCH /consentRequest/{ID} PISP response requires user input
+      // on the PISP side we need to make this a longer time.
+      // todo: figure out what time is adequate and make a new variable to store it
+      .wait(this.config.requestProcessingTimeoutSeconds * 1000)
+
+    // catch any unplanned errors and notify PISP
+    try {
+      await waitOnAuthToken
```

Comment:
I would put whole block L221-256 here so whole deferredJob code in in try/catch

Reply:
I like how this reads a bit better. Is there something functional I'm overlooking doing it this way?